runtime/src/main/java/org/corfudb/runtime/collections/CorfuStore.java

```diff
@@ -89,6 +112,7 @@ private Timestamp getTimestamp() {
      * @throws InvocationTargetException Thrown if key/value class are not protobuf classes.
      * @throws IllegalAccessException    Thrown if key/value class are not protobuf classes.
      */
+    @Deprecated
```

Comment:
This can cause an explosion in the build tool chain until all the 1000+ consumers are migrated over to the new one.

Reply:
Hi Sundar,
 - external API didn't change
 - the deprecated api can stay for a long time, we should be fine, because this code is internal, am I correct?
 - TableDescriptor & FullyQualifiedTableName brings two advantages:
     - reduce the amount of parameters, which reduces mental complexity or reading 8 parameters in methods signatures and makes code easier to understand with fine grained names
     - the classes aggregate functionality they are in charge of, which makes it easer to write unit tests and helps to get rid of copy pasting a bunch of parameters. For instance, we just can use an instance of FullyQualifiedTableName and get stream id, fullyqualified table name and stream name from one place (don't forget about easy unit testing of those name builders), rather than passing 2-3 params every single time and copy-pasting CorfuRuntime.getStreamId(UUID.fromByutes(TableRegistry.getFqdn(tableName, namespace))) which is speards across the entire system, and makes it much harder to comprehend what's going on in the code and build a mental picture of the code.

That was my main motivation when I did my refactoring, you can see additional details in #3979